pkgs/servers/web-apps/plausible/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib, stdenv, beamPackages, fetchFromGitHub, glibcLocales, cacert
+, mkYarnModules, nodejs, fetchpatch, nixosTests
+}:
+
+let
+  pname = "plausible";
+  version = "1.3.0";
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "plausible";
+    repo = "analytics";
+    rev = "v${version}";
+    sha256 = "03lm1f29gwwixnhgjish5bhi3m73qyp71ns2sczdnwnbhrw61zps";
+  };
+
+  mixDeps = beamPackages.fetchMixDeps {
+    pname = "${pname}-deps";
+    inherit src version;
+    sha256 = "sha256-66zSdYmis3UnbdLkPi649RbPbMPI5gVeFlaMekOy5CQ=";
+  };
+
+  yarnDeps = mkYarnModules {
+    pname = "${pname}-yarn-deps";
+    inherit version;
+    packageJSON = ./package.json;
+    yarnNix = ./yarn.nix;
+    yarnLock = ./yarn.lock;
+    preBuild = ''
+      mkdir -p tmp/deps
+      cp -r ${mixDeps}/phoenix tmp/deps/phoenix
+      cp -r ${mixDeps}/phoenix_html tmp/deps/phoenix_html
+    '';
+    postBuild = ''
+      echo 'module.exports = {}' > $out/node_modules/flatpickr/dist/postcss.config.js
+    '';
+  };
+in beamPackages.mixRelease {
+  inherit pname version src mixDeps;
+
+  nativeBuildInputs = [ nodejs ];
+
+  patches = [
+    # Allow socket-authentication against postgresql. Upstream PR is
+    # https://github.com/plausible/analytics/pull/1052
+    (fetchpatch {
+      url = "https://github.com/Ma27/analytics/commit/f2ee5892a6c3e1a861d69ed30cac43e05e9cd36f.patch";
+      sha256 = "sha256-JvJ7xlGw+tHtWje+jiQChVC4KTyqqdq2q+MIcOv/k1o=";
+    })
+  ];
+
+  passthru = {
+    tests = { inherit (nixosTests) plausible; };
+    updateScript = ./update.sh;
+  };
+
+  postPatch = ''
+    # Without this modification, tzdata tries to write in its store-path:
+    # https://github.com/lau/tzdata#data-directory-and-releases
+    echo 'config :tzdata, :data_dir, (System.get_env("PLAUSIBLE_TZDATA") || "/tmp/plausible_tzdata")' \
+      >> config/config.exs
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    mkdir -p $out
+    ln -sf ${yarnDeps}/node_modules assets/node_modules
+    mix deps.compile --path $out --no-deps-check
```

Comment:
deps.compile should already be done in the configure phase.

I'm not sure you need to override the whole buildPhase here.

Probably just adding a `postConfigure` steps with the yarn stuff should be enough
something like
```
ln -sf ${yarnDeps}/node_modules assets/node_modules
npm run deploy --prefix ./assets
mix phx.digest --no-deps-check
```

Reply:
> I'm not sure you need to override the whole buildPhase here.

Tbh, I mostly redid the steps from the `Dockerfile` (and IIRC `mixRelease` didn't exist when I wrote a draft in Februrary). But yeah, I should've taken a closer look here, sorry! Will do next week.

But `buildPhase` (or `postBuild` - idk) is at least needed for the `npm` stuff.

Reply:
yes, to run phx.digest you need postBuild, you're right.